### PR TITLE
Hide all future tag expirations in the timeline, by default

### PR DESCRIPTION
**Pull request by josephschorr**
Fixes https://issues.redhat.com/browse/PROJQUAY-220

Screen Shot 2020-02-14 at 2 19 49 PM

### CODE REVIEW

File: static/directives/repo-tag-history.html [hunk 1]
```diff
@@ -20,7 +21,7 @@
         </thead>
         <tbody>
           <tr style="height: 20px;"><td colspan="4"></td></tr>
-          <tr ng-repeat="entry in historyEntries" class="history-entry" ng-class="getEntryClasses(entry, filter)">
+          <tr ng-repeat="entry in historyEntries" class="history-entry" ng-class="getEntryClasses(entry, filter, options.showFuture)">
```
Comment (alecmerdler) [inline static/directives/repo-tag-history.html]: The 3rd argument isn't being used in the function itself (only accepts 2 parameters).

File: static/js/directives/ui/repo-tag-history.js [hunk 2]
```diff
@@ -150,6 +163,11 @@ angular.module('quay').directive('repoTagHistory', function () {
         }
       };
 
+      $scope.isFuture = function(entry) {
+        if (!entry) { return false; }
+        return entry.time >= new Date().getTime();
+      };
+
       $scope.getEntryClasses = function(entry, historyFilter) {
```
Comment (alecmerdler) [inline static/js/directives/ui/repo-tag-history.js]: You pass 3 arguments in the AngularJS template, but there are only 2 parameters here.

File: static/js/directives/ui/repo-tag-history.js [hunk 3]
```diff
@@ -31,12 +35,17 @@ angular.module('quay').directive('repoTagHistory', function () {
...
 
         ApiService.listRepoTags(null, params).then(function(resp) {
+          $scope.cachedFullTags = resp.tags;
           processTags(resp.tags);
         });
       };
 
       $scope.$watch('isEnabled', loadTimeline);
       $scope.$watch('repositoryTags', loadTimeline);
+      $scope.$watch('options.showFuture', function() {
```
Comment (alecmerdler) [inline static/js/directives/ui/repo-tag-history.js]: Prefer not to use `$scope.$watch()`, but this is fine for now.

File: static/directives/repo-tag-history.html (hunk 1 above)
Comment (josephschorr) [reply static/directives/repo-tag-history.html]: Whoops... forgot to remove that. Done.

File: static/js/directives/ui/repo-tag-history.js (hunk 3 above)
Comment (josephschorr) [reply static/js/directives/ui/repo-tag-history.js]: Angular 1!

**Pull request merged**